django/db/models/fields/files.py

```diff
@@ -226,7 +227,10 @@ def __init__(self, verbose_name=None, name=None, upload_to='', storage=None, **k
         self.storage = storage or default_storage
 
         if callable(self.storage):
-            self.storage = self.storage()
+            check_storage = self.storage()
+            if not issubclass(check_storage, Storage):
+                raise ImproperlyConfigured("%s.storage must be a subclass of %s.%s ()" % (self.__class__.__name__, Storage.__module__, Storage.__name__))
+            self.storage = check_storage
```

Comment:
* You need to use `isinstance` here not `issubclass`. 
* I don't think we need the temporary variable. Just assign to `self.storage` as before and then check that.

Reply:
Yeah, it is instantiated, I need to adjust a bit still. It has to be issubclass though.